js-timezone-selector-plugin/src/selector/index.tsx

```diff
@@ -5,35 +5,39 @@ import Button from "./Button";
 import OptionsContainer from "./OptionsContainer";
 import { DEFAULT_VALUE } from './utils';
 
-const Selector = ({ className = "", position = "bottom", onChange = (selectedValue) => { } }) => {
+const Selector = ({ className = "", position = "bottom", onChange = (selectedValue) => { }, elementId = "timezone-selector" }) => {
   const [selectedValue, setSelectedValue] = useState(DEFAULT_VALUE);
   const [showOptions, setShowOptions] = useState(false);
   const top = position === "top";
 
-  const addOutsideClickListener = () => {
-    window.addEventListener('click', function (e) {
-      if (!(document.getElementById('timezoneSelector')?.contains(e.target))) {
-        setShowOptions(false)
+  useEffect(() => {
+    const addOutsideClickListener = (e) => {
```

Comment:
`add` doesn't make sense here anymore. It's a callback function, let's follow the neeto naming standards here

```suggestion
    const handleOutsideClick = (e) => {
```